### PR TITLE
GEODE-5298: User Guide - Document Transactions + Functions Limitation

### PR DESCRIPTION
**Pull request by davebarnes97**
Documented the requirement that a function within a transaction must operate on colocated data.
Reinstated some cross-ref targets for improved navigation.

### CODE REVIEW

File: geode-docs/developing/transactions/working_with_transactions.html.md.erb [hunk 1]
```diff
@@ -76,37 +76,48 @@ r.put("stringBuf", s);
...
-If you are suspending and resuming a transaction with multiple function calls, all functions in the transaction must execute on the same member.
+  - **When a function contains multiple transactions,** the transactions must be sequential, not nested.
+  - **When a transaction contains a function,** the function must operate only on keys that are colocated with the keys in the transaction,
+as described in [Data Location for Cache Transactions](data_location_cache_transactions.html). This implies:
+
+      - A function within a transaction must not specify execution on a particular server or member. 
+<%=vars.product_name%> throws `UnsupportedOperationException` if `FunctionService` calls `onMember()`, `onServer()`, 
+or similar methods from within a transaction.
+
+      - A function within a transaction can request a particular region, that is, the `FunctionService.onRegion()` method is permitted.
```
Comment (joeymcallister) [inline geode-docs/developing/transactions/working_with_transactions.html.md.erb]: Change "region, that is, the" to "region—that is, the"

**Comment by davebarnes97** [comment]
Oops - thanks for catching the bad JIRA number, Karen. I'll correct that.

File: geode-docs/developing/transactions/working_with_transactions.html.md.erb [hunk 2]
```diff
@@ -76,37 +76,48 @@ r.put("stringBuf", s);
...
+## <a id="concept_ihn_zmt_vk" class="no-quick-link"></a>Transactions and Functions
 
-You can run a function from inside a transaction and you can nest a transaction within a function, as long as your combination of functions and transactions does not result in nested transactions. See [Function Execution](../function_exec/chapter_overview.html) for more about functions.
+A function may contain transactions, and a transaction may contain functions, as long as you observe these restrictions:
 
-A single transaction may contain multiple functions.
+  - **Your combination of functions and transactions must not result in nested transactions.**
 
-If you are suspending and resuming a transaction with multiple function calls, all functions in the transaction must execute on the same member.
+  - **When a function contains multiple transactions,** the transactions must be sequential, not nested.
```
Comment (sbawaska) [inline geode-docs/developing/transactions/working_with_transactions.html.md.erb]: I think this is the same as the bullet above

Comment (davebarnes97) [reply geode-docs/developing/transactions/working_with_transactions.html.md.erb]: You're right, it's redundant. Will delete.

**Comment by davebarnes97** [comment]
Approvals complete. Merging manually in order to correct a typo in the original commit message (incorrect JIRA ticket number).

**Pull request closed without merging**